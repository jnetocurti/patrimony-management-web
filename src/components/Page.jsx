import React, { forwardRef } from 'react';
import { Helmet } from 'react-helmet';

import PropTypes from 'prop-types';

const Page = forwardRef(({ children, title = '', ...rest }, ref) => {
  return (
    <div ref={ref} {...rest}>
      <Helmet>
        <title>{title}</title>
      </Helmet>
      {children}
    </div>
  );
});

Page.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Page;
