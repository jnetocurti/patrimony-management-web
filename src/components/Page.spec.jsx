import React, { Fragment } from 'react';

import { shallow } from 'enzyme';

import Page from './Page';

describe('src/components/Page', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Page>
        <></>
      </Page>
    );
    expect(component).toMatchSnapshot();
  });
});
