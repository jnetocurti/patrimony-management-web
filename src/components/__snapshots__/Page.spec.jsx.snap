// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Page should render correctly 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title />
  </HelmetWrapper>
</div>
`;
